### PR TITLE
Fix a memory leak.

### PR DESCRIPTION
**Pull request by bangerth**
While there, also clean up around it.

### CODE REVIEW

**Comment by bangerth** [comment]
Part of #2375.

File: source/boundary_temperature/dynamic_core.cc [hunk 1]
```diff
@@ -303,21 +303,24 @@ namespace aspect
...
       data_OES.clear();
       if (name_OES.size()==0) return;
-      std::istringstream in(Utilities::read_and_distribute_file_content(name_OES.c_str(), this->get_mpi_communicator()));
+      std::istringstream in(Utilities::read_and_distribute_file_content(name_OES.c_str(),
+                                                                        this->get_mpi_communicator()));
       if (in.good())
         {
           str_data_OES data_read;
-          const int buff_size = 1024;
-          char *line = new char [buff_size];
```
Comment (tjhei) [inline source/boundary_temperature/dynamic_core.cc]: important to do a dynamic allocation here. ;-)

**Comment by tjhei** [comment]
:+1:

File: source/boundary_temperature/dynamic_core.cc (hunk 1 above)
Comment (bangerth) [reply source/boundary_temperature/dynamic_core.cc]: Yes, one could have done without that here :-)

**Pull request merged**